fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/LoanDelinquencyDomainServiceImpl.java

```diff
@@ -66,10 +69,11 @@ public CollectionData getOverdueCollectionData(final Loan loan) {
 
                         amountAvailable = installment.getTotalPaid(loanCurrency).getAmount();
 
-                        for (LoanTransactionToRepaymentScheduleMapping mappingInstallment : installment
-                                .getLoanTransactionToRepaymentScheduleMappings()) {
-                            final LoanTransaction loanTransaction = mappingInstallment.getLoanTransaction();
-                            if (loanTransaction.isChargeback()) {
+                        boolean isLatestInstallment = installment.equals(latestInstallment);
+                        for (LoanTransaction loanTransaction : chargebackTransactions) {
+                            if (!loanTransaction.getTransactionDate().isBefore(installment.getFromDate()) && ((!isLatestInstallment
```

Comment:
We have many many places with logic like this and I always feel they are a bit hard to read. Do you think it'd make sense to extract these individual condition pieces into variables before the if statement? Like this:

```
boolean isTransactionBeforeInstallmentStart = loanTransaction.getTransactionDate().isBefore(installment.getFromDate();
....
if (!isTransactionBeforeInstallmentStart && ...) {
    ...
}
```
Also, positive conditions are usually better, please evaluate whether we can get rid of the negate operator and rename the variable accordingly.